app/models/katello/glue/provider.rb

```diff
@@ -144,14 +150,39 @@ def import_product(product_json)
         if product&.redhat?
           product.update!(:name => product_json['name']) unless product.name == product_json['name']
         elsif product.nil?
-          product = Glue::Candlepin::Product.import_from_cp(product_json, organization)
+          # we have an orphaned Red Hat product
+          product = Glue::Candlepin::Product.import_redhat_product_from_cp(product_json, organization)
         end
         product
       end
 
-      def index_subscriptions
-        Katello::Subscription.import_all(organization)
-        Katello::Pool.import_all(organization, false)
+      def orphaned_custom_product?(product_json, organization)
```

Comment:
Now that these are instance methods I think you already have access to `organization` and don't need to send it as a param